src/components/dialogs/parameters/single-line-diagram-parameters.js

```diff
@@ -35,19 +34,18 @@ export const useGetAvailableComponentLibraries = (user) => {
 };
 
 export const SingleLineDiagramParameters = ({
-    hideParameters,
-    componentLibraries,
+    hideParameters, //function
+    componentLibraries, //array<string>
 }) => {
     const classes = useStyles();
-
-    const [diagonalLabelLocal, handleChangeDiagonalLabel] =
-        useParameterState(PARAM_DIAGONAL_LABEL);
-    const [centerLabelLocal, handleChangeCenterLabel] =
-        useParameterState(PARAM_CENTER_LABEL);
-    const [substationLayoutLocal, handleChangeSubstationLayout] =
-        useParameterState(PARAM_SUBSTATION_LAYOUT);
-    const [componentLibraryLocal, handleChangeComponentLibrary] =
-        useParameterState(PARAM_COMPONENT_LIBRARY);
+    const cmpntLibs = useMemo(
```

Comment:
I have reduce "component" to its acronym "cmpnt" because I found it long to write. For a "complete" one, It would be "componentLibrariesStates", as [useMemo](https://react.dev/reference/react/useMemo) do a cache of the rendering of the components.